pkg/reconciler/buildrun/resources/sources/bundle.go

```diff
@@ -54,3 +62,15 @@ func AppendBundleStep(
 
 	taskSpec.Steps = append(taskSpec.Steps, bundleStep)
 }
+
+// AppendBundleResult append bundle source result to build run
+func AppendBundleResult(buildRun *build.BuildRun, name string, results []pipeline.TaskRunResult) {
+	imageDigest := findResultValue(results, fmt.Sprintf("%s-source-%s-image-digest", prefixParamsResultsVolumes, name))
+
+	buildRun.Status.Sources = append(buildRun.Status.Sources, build.SourceResult{
```

Comment:
slight tweak, make it a strings.TrimSpaces(imageDigest) != ""

Reply:
In order to avoid setting empty results, I am already checking the length of `TaskRunResults` before calling the function that surfacing the results. [See here](https://github.com/shipwright-io/build/pull/871/files#diff-92a124fc33b3ec7d38392c0b313144f552c7987f35bb1aad07ea355020ad32e3R300-R303).

Please let me know if changes are required.

Reply:
Yes, because eventually we might have multiple source steps and in case of a failure, step 1 might succeed (and produce results) and step 2 fails which produces therefore no success result but maybe [error results that we plan to introduce as well](https://github.com/shipwright-io/community/pull/30).